pkg/reconciler/instances/istio/action.go

```diff
@@ -52,31 +40,27 @@ func (a *UninstallAction) Run(context *service.ActionContext) error {
 	if err != nil {
 		return err
 	}
-	if canUninstall(istioStatus) {
-		component := chart.NewComponentBuilder(context.Task.Version, context.Task.Component).
-			WithNamespace(istioNamespace).
-			WithProfile(context.Task.Profile).
-			WithConfiguration(context.Task.Configuration).Build()
-		istioManifest, err := context.ChartProvider.RenderManifest(component)
-		if err != nil {
-			return err
-		}
-		// Before removing istio himself, undeploy all related objects like dashboards
-		err = unDeployIstioRelatedResources(context.Context, istioManifest.Manifest, context.KubeClient, context.Logger)
-		if err != nil {
-			return err
-		}
-		err = performer.Uninstall(context.KubeClient, istioStatus.TargetVersion, context.Logger)
-		if err != nil {
-			return errors.Wrap(err, "Could not uninstall istio")
-		}
-		context.Logger.Infof("Istio successfully uninstalled")
-	} else {
-		context.Logger.Warnf("Istio is not installed, can not uninstall it")
+
+	if isMismatchPresent(istioStatus) {
+		errorMessage := fmt.Sprintf("Istio components version mismatch detected: pilot version: %s, data plane version: %s", istioStatus.PilotVersion, istioStatus.DataPlaneVersion)
+		return errors.New(errorMessage)
 	}
+
+	if !isClientCompatibleWithTargetVersion(istioStatus) {
+		return errors.New(fmt.Sprintf("Istio could not be updated since the binary version: %s is not compatible with the target version: %s - the difference between versions exceeds one minor version", istioStatus.ClientVersion, istioStatus.TargetVersion))
+	}
+
 	return nil
 }
 
+type ReconcileAction struct {
```

Comment:
As above. What about naming it: `IstioHandlingMainAction` or `IstioInstallUpgradeMainAction` or something similar?

Reply:
Yup, so this way we would simply omit the first word